dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java

```diff
@@ -0,0 +1,126 @@
+import com.google.common.io.Resources;
+import com.google.edwmigration.dumper.application.dumper.io.OutputHandle;
+import com.google.edwmigration.dumper.application.dumper.task.Task;
+import com.google.edwmigration.dumper.application.dumper.task.TaskCategory;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.Writer;
+import java.net.URL;
+import java.time.Duration;
+import org.apache.commons.exec.CommandLine;
+import org.apache.commons.exec.DefaultExecutor;
+import org.apache.commons.exec.ExecuteException;
+import org.apache.commons.exec.ExecuteWatchdog;
+import org.apache.commons.exec.PumpStreamHandler;
+
+public class BashTask implements Task<Void> {
+
+  private static final Duration SCRIPT_TIMEOUT = Duration.ofMinutes(5);
+
+  private final String scriptName;
+
+  public BashTask(String scriptName) {
+    this.scriptName = scriptName;
+  }
+
+  @Override
+  public String getTargetPath() {
+    return scriptName + ".out";
+  }
+
+  private void doRun(ByteSink outputSink, ByteSink errorSink, ByteSink exitStatusSink)
+      throws IOException {
+    String scriptFilename = scriptName + ".sh";
+    URL resourceUrl = Resources.getResource("hadoop-scripts/" + scriptFilename);
+    byte[] scriptBody = Resources.toByteArray(resourceUrl);
+    File scriptDir = new File("dwh-migration-tools-tmp");
+    scriptDir.mkdirs();
+    File scriptFile = new File(scriptDir, scriptFilename);
+    Files.write(scriptBody, scriptFile);
+    scriptFile.setExecutable(true);
+    CommandLine cmdLine = CommandLine.parse("/bin/bash " + scriptFile.getAbsolutePath());
+    DefaultExecutor executor = DefaultExecutor.builder().get();
+    executor.setExitValue(1);
+    ExecuteWatchdog watchdog = ExecuteWatchdog.builder().setTimeout(SCRIPT_TIMEOUT).get();
+    executor.setWatchdog(watchdog);
+    try (OutputStream outputStream = outputSink.openBufferedStream();
+        OutputStream errorStream = errorSink.openBufferedStream()) {
+      PumpStreamHandler streamHandler = new PumpStreamHandler(outputStream, errorStream);
+      executor.setStreamHandler(streamHandler);
+      int exitValue = executor.execute(cmdLine);
+      try (Writer wr = exitStatusSink.asCharSink(UTF_8).openBufferedStream()) {
+        wr.write("" + exitValue);
+      }
+    } catch (ExecuteException e) {
+      try (Writer wr = exitStatusSink.asCharSink(UTF_8).openBufferedStream()) {
+        wr.write("" + e.getExitValue());
+      }
+      if (e.getExitValue() != 0) {
+        throw e;
+      }
+    }
+  }
+
+  @Override
+  public Void run(TaskRunContext context) throws Exception {
+    OutputHandle outputHandle = createOutputHandle(context, scriptName + ".out");
+    OutputHandle errorOutputHandle = createOutputHandle(context, scriptName + ".err");
+    OutputHandle exitStatusOutputHandle = createOutputHandle(context, scriptName + ".exit-status");
```

Comment:
What is the use case of `.exit-status` file? Do you plan to read it in assessment? Would it be possible to write error code to `.err` file? We can then assume if `.err` is empty there are no errors (status code is 0)